### PR TITLE
Fix several dead-links

### PR DESCRIPTION
**Pull request by badboy**
This fixes some dead links and moves the activity stream dataset docs into obsolete.

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -151,10 +150,11 @@
     - [Metrics Standardization and Policy](metrics/policy.md)
   - [Legacy Census Metrics](concepts/censuses.md)
   - [Obsolete Datasets](datasets/obsolete.md)
+    - [Activity Stream](datasets/obsolete/activity-stream/reference.md)
```
Comment (badboy) [inline src/SUMMARY.md:153]: @chutten as promised the move here.

File: src/SUMMARY.md [hunk 2]
```diff
@@ -151,10 +150,11 @@
...
     - [Attitudes Daily](datasets/obsolete/attitudes_daily/reference.md)
     - [Churn](datasets/obsolete/churn/reference.md)
-    - [Client Count](datasets/obsolete/client_count/reference.md)
     - [Client Count Daily](datasets/obsolete/client_count_daily/reference.md)
+    - [Client Count](datasets/obsolete/client_count/reference.md)
```
Comment (badboy) [inline src/SUMMARY.md:157]: Oh, I re-ordered those things by alphabet.

File: src/datasets/reference.md [hunk 3]
```diff
@@ -6,8 +6,7 @@ Instead, identify a dataset you'd like to understand better and read through
 the relevant documentation.
 After reading the tutorial, you should know all you need about the dataset.
 
-Also see more detailed [per-table docs](https://mozilla.github.io/bigquery-etl/mozdata/introduction/)
-that are generated alongside ETL code in the `bigquery-etl` repository.
+Detailed per-table docs are available [in the Data Catalog](https://mozilla.acryl.io/).
```
Comment (badboy) [inline src/datasets/reference.md:9]: @ANich ok like this?

Comment (ANich) [reply src/datasets/reference.md:9]: Yeah this is okay, I'm going to write a longer page on this so I might update this in that PR 👍

**Review by ANich — APPROVED** [review]

**Review by chutten — APPROVED** [review]

**Pull request merged**